phpBB/phpbb/group/helper.php

```diff
@@ -13,19 +13,73 @@
 
 namespace phpbb\group;
 
+use phpbb\auth\auth;
+use phpbb\cache\service as cache;
+use phpbb\config\config;
+use phpbb\language\language;
+use phpbb\event\dispatcher_interface;
+use phpbb\path_helper;
+use phpbb\user;
+
 class helper
 {
-	/** @var  \phpbb\language\language */
+	/** @var auth */
+	protected $auth;
+
+	/** @var cache */
+	protected $cache;
+
+	/** @var config */
+	protected $config;
+
+	/** @var language */
 	protected $language;
 
+	/** @var dispatcher_interface */
+	protected $dispatcher;
+
+	/** @var path_helper */
+	protected $path_helper;
+
+	/** @var user */
+	protected $user;
+
+	/** @var string phpBB root path */
+	protected $phpbb_root_path;
+
+	/** @var array Return templates for a group name string */
+	protected $name_strings;
+
 	/**
 	 * Constructor
 	 *
-	 * @param \phpbb\language\language $language	Language object
+	 * @param auth					$auth			Authentication object
+	 * @param cache					$cache			Cache service object
+	 * @param config				$config			Configuration object
+	 * @param language				$language		Language object
+	 * @param dispatcher_interface	$dispatcher		Event dispatcher object
+	 * @param path_helper			$path_helper	Path helper object
+	 * @param user					$user			User object
 	 */
-	public function __construct(\phpbb\language\language $language)
+	public function __construct(auth $auth, cache $cache, config $config, language $language, dispatcher_interface $dispatcher, path_helper $path_helper, user $user)
 	{
+		$this->auth = $auth;
+		$this->cache = $cache;
+		$this->config = $config;
 		$this->language = $language;
+		$this->dispatcher = $dispatcher;
+		$this->path_helper = $path_helper;
+		$this->user = $user;
+
+		$this->phpbb_root_path = $path_helper->get_phpbb_root_path();
+
+		$this->name_strings = array(
+			'base_url'				=> append_sid("{$path_helper->get_phpbb_root_path()}memberlist.{$path_helper->get_php_ext()}", 'mode=group&amp;g={GROUP_ID}'),
```

Comment:
Please don't call this in the constructor but actually use append_sid() when you need it. The container will automatically share services and therefore might end up sharing this URL with the SID across users.

Reply:
Changed it so assignment is still as a class member, but `append_sid()` is only called when required.